### PR TITLE
Add pcre2 regex

### PR DESCRIPTION
**Pull request by dylan-conway**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -394,7 +394,9 @@ MINIMUM_ARCHIVE_FILES = -L$(BUN_DEPS_OUT_DIR) \
 	-lssl \
 	-lcrypto \
 	-llolhtml \
-	$(BUN_DEPS_OUT_DIR)/libbacktrace.a
+	$(BUN_DEPS_OUT_DIR)/libbacktrace.a \
+	$(BUN_DEPS_OUT_DIR)/libpcre2-16.a \
```
Comment (Jarred-Sumner) [inline Makefile]: I wonder if we need both of these

File: src/bun.js/bindings/PCRE2RegExp.cpp [hunk 2]
```diff
@@ -0,0 +1,790 @@
...
+    if (jsString->length() == 0) {
+        return to16Bit(defaultValue);
+    }
+
+    return to16Bit(jsString, globalObject);
+}
+
+static inline bool is16BitLineTerminator(UChar c)
+{
+    return c == '\r' || c == '\n' || (c & ~1) == 0x2028;
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/PCRE2RegExp.cpp:51]: There are technically about 10 more of these but I assume this was copied from RegExpPrototype.cpp in WebKit so it's probably fine

File: src/bun.js/bindings/PCRE2RegExp.cpp [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+    }
+
+    // static void analyzeHeap(JSCell*, JSC::HeapAnalyzer&);
+
+    const WTF::String& flagsString() const { return m_flagsString; }
+    void setFlagsString(const WTF::String& flagsString) { m_flagsString = flagsString; }
+    const WTF::String& patternString() const { return m_patternString; }
+    void setPatternString(const WTF::String& patternString) { m_patternString = patternString; }
+
+    pcre2_code_16* m_regExpCode = NULL;
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/PCRE2RegExp.cpp:264]: do we need to free this in a destructor?

File: src/bun.js/bindings/PCRE2RegExp.cpp [hunk 4]
```diff
@@ -0,0 +1,790 @@
...
+    return JSValue::encode(jsBoolean(thisValue->flagsString().contains('d') ? true : false));
+}
+
+JSC_DEFINE_CUSTOM_GETTER(pcre2RegExpProtoGetterIgnoreCase, (JSGlobalObject *globalObject, EncodedJSValue encodedThis, PropertyName))
+{
+    auto *thisValue = jsDynamicCast<PCRE2RegExp*>(JSValue::decode(encodedThis));
+    if (UNLIKELY(!thisValue)) {
+        return JSValue::encode(jsUndefined());
+    }
+    return JSValue::encode(jsBoolean(thisValue->flagsString().contains('i') ? true : false));
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/PCRE2RegExp.cpp]: the ? true : false are redundant here

File: src/bun.js/bindings/PCRE2RegExp.cpp [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+        WTF::StringBuilder errorMessage = WTF::StringBuilder();
+        errorMessage.append("Invalid regular expression: ");
+        errorMessage.append(reinterpret_cast<const char16_t*>(buffer));
+        errorMessage.append(" at offset: ");
+        errorMessage.append(errorOffset);
+        throwScope.throwException(globalObject, createSyntaxError(globalObject, errorMessage.toString()));
+        return JSValue::encode({});
+    }
+
+    pcre2_code_free_16(thisRegExp->m_regExpCode);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/PCRE2RegExp.cpp:475]: Is there a case where `m_regExpCode` can be null here?

File: src/bun.js/bindings/PCRE2RegExp.cpp [hunk 6]
```diff
@@ -0,0 +1,790 @@
...
+                pcre2_match_data_free_16(matchData);
+                return JSValue::encode(jsNull());
+            }
+        
+            memcpy(ptr, substringStart, substringLength * sizeof(UChar));;
+        }
+
+        result->putDirectIndex(globalObject, i, jsString(vm, outString));
+
+        JSArray* indices = constructEmptyArray(globalObject, nullptr, 2);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/PCRE2RegExp.cpp:615]: There's a faster way to construct short arrays like this, but I think it's fine to ignore for now

File: src/bun.js/bindings/sqlite/JSSQLStatement.cpp [hunk 7]
```diff
@@ -465,14 +458,6 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementDeserialize, (JSC::JSGlobalObject * lexic
         return JSValue::encode(JSC::jsUndefined());
     }
 
-#if LAZY_LOAD_SQLITE
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/sqlite/JSSQLStatement.cpp:468]: Please pull the latest changes, this change reverts a recent commit.

File: Makefile (hunk 1 above)
Comment (dylan-conway) [reply Makefile]: removed libpcre2-posix

File: src/bun.js/bindings/PCRE2RegExp.cpp (hunk 5 above)
Comment (dylan-conway) [reply src/bun.js/bindings/PCRE2RegExp.cpp:475]: If `m_regExpCode` is null, `pcre2_code_free_16()` will do nothing and return, so it should be okay to free each time

**Comment by dylan-conway** [comment]
Switched to oniguruma for variable length look-behinds #1329

**Pull request closed without merging**